src/main/java/pl/javastart/task/SpecialOffer.java

```diff
@@ -0,0 +1,26 @@
+package pl.javastart.task;
+
+public class SpecialOffer {
+    Product product;
+    String description;
+    String duration;
+    double discount;
+
+    public SpecialOffer(Product product, String description, String duration, double discount) {
+        this.product = product;
+        this.description = description;
+        this.duration = duration;
+        this.discount = discount;
+    }
+
+    void setDiscount() {
+        product.price -= product.price * discount;
```

Comment:
Nie zmieniaj ceny produktu na stałe! Tracisz wtedy informację ile kosztował pierwotnie. W przypadku wyświetlania możesz obliczać cenę 'na bieżąco", bez jej zmiany. Wystarczy, że utworzysz to tego celu nową zmienną.

Poza tym, przy takim zastosowaniu w setterze "naliczasz" rabat, a przy ustawianiu przez konstruktor nie. Najlepiej usuń ten setter i będzie ok, bo na tym etapie nie oczekujemy wyświetlania ceny produktu po rabacie.